ownCloud/Release Notes/ReleaseNotesHostViewController.swift

```diff
@@ -0,0 +1,211 @@
+				footerLabel.topAnchor.constraint(equalTo: bottomView.topAnchor, constant: smallPadding),
+				footerLabel.bottomAnchor.constraint(equalTo: proceedButton.topAnchor, constant: padding * -1)
+			])
+
+			NSLayoutConstraint.activate([
+				proceedButton.leadingAnchor.constraint(equalTo: bottomView.leadingAnchor, constant: padding),
+				proceedButton.trailingAnchor.constraint(equalTo: bottomView.trailingAnchor, constant: padding * -1),
+				proceedButton.heightAnchor.constraint(equalToConstant: buttonHeight),
+				proceedButton.bottomAnchor.constraint(equalTo: bottomView.bottomAnchor, constant: smallPadding * -1)
+			])
+
+			NSLayoutConstraint.activate([
+				containerView.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor),
+				containerView.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor),
+				containerView.topAnchor.constraint(equalTo: headerView.bottomAnchor),
+				containerView.bottomAnchor.constraint(equalTo: bottomView.topAnchor)
+			])
+		}
+	}
+
+	deinit {
+		Theme.shared.unregister(client: self)
+	}
+
+	@objc func dismissView() {
+		self.dismiss(animated: true, completion: nil)
+	}
+}
+
+// MARK: - Themeable implementation
+extension ReleaseNotesHostViewController : Themeable {
+	func applyThemeCollection(theme: Theme, collection: ThemeCollection, event: ThemeEvent) {
+
+		self.view.backgroundColor = collection.tableBackgroundColor
+		titleLabel.applyThemeCollection(collection, itemStyle: .logo)
+		proceedButton.backgroundColor = collection.neutralColors.normal.background
+		proceedButton.setTitleColor(collection.neutralColors.normal.foreground, for: .normal)
+		footerLabel.textColor = collection.tableRowColors.labelColor
+	}
+}
+
+class ReleaseNotesDatasource : NSObject, OCClassSettingsUserPreferencesSupport {
+
+	var shouldShowReleaseNotes: Bool {
+		if let lastSeenReleaseNotesVersion = self.classSetting(forOCClassSettingsKey: .lastSeenReleaseNotesVersion) as? String {
+
+			if lastSeenReleaseNotesVersion.compare(VendorServices.shared.appVersion, options: .numeric) == .orderedDescending || lastSeenReleaseNotesVersion.compare(VendorServices.shared.appVersion, options: .numeric) == .orderedSame {
+				return false
+			}
+
+			if let path = Bundle.main.path(forResource: "ReleaseNotes", ofType: "plist") {
+				if let releaseNotesValues = NSDictionary(contentsOfFile: path), let versionsValues = releaseNotesValues["Versions"] as? NSArray {
+
+					let relevantReleaseNotes = versionsValues.filter {
+						if let version = ($0 as AnyObject)["Version"] as? String, version.compare(VendorServices.shared.appVersion, options: .numeric) == .orderedDescending {
+							return false
+						}
+
+						return true
+					}
+
+					if relevantReleaseNotes.count > 0 {
+						return true
+					}
+				}
+			}
+
+			return false
+		}
+
+		return true
+	}
+
+	func releaseNotes(for version: String) -> [[String:Any]]? {
+		if let path = Bundle.main.path(forResource: "ReleaseNotes", ofType: "plist") {
```

Comment:
@hosy Looking at code in lines 154-172, I think there is some potential to reduce code duplication...

Reply:
removed one if statement